library/std/src/sys/pal/windows/thread.rs

```diff
@@ -71,6 +71,28 @@ impl Thread {
         };
     }
 
+    pub fn get_name() -> Option<CString> {
+        unsafe {
+            let mut ptr = core::ptr::null_mut();
+            let result = c::GetThreadDescription(c::GetCurrentThread(), &mut ptr);
+            if result < 0 {
+                return None;
+            }
+            let name = String::from_utf16_lossy({
+                let mut len = 0;
+                while *ptr.add(len) != 0 {
+                    len += 1;
+                }
+                core::slice::from_raw_parts(ptr, len)
+            })
+            .into_bytes();
+            // Attempt to free the memory.
+            // This should never fail but if it does then there's not much we can do about it.
+            debug_assert!(c::LocalFree(ptr.cast::<c_void>()).is_null());
```

Comment:
Shouldn't this call `LocalFree` even if debug assertions aren't enabled?

Reply:
Oh, doh! Yeah, I was trying to be too clever by half and really messed that up.